openslides/users/static/js/users/site.js

```diff
@@ -338,168 +353,258 @@ angular.module('OpenSlidesApp.users.site', ['OpenSlidesApp.users'])
-            user.title = obj[i].titel;
-            user.first_name = obj[i].first_name;
-            user.last_name = obj[i].last_name;
-            user.structure_level = obj[i].structure_level;
-            user.groups = [];
-            if (obj[i].groups !== '') {
-                var groups = obj[i].groups.replace('"','').split(",");
-                groups.forEach(function(group) {
-                    user.groups.push(group);
-                    console.log(group);
-                });
+        };
+    }
+])
+
+.controller('UserUpdateCtrl', [
+    '$scope',
+    '$state',
+    'User',
+    'user',
+    'Group',
+    function($scope, $state, User, user, Group) {
+        Group.bindAll({where: {id: {'>': 2}}}, $scope, 'groups');
+        $scope.user = user;  // autoupdate is not activated
+        $scope.save = function (user) {
+            if (!user.groups) {
+                user.groups = [];
             }
-            user.comment = obj[i].comment;
-            User.create(user).then(
+            User.save(user).then(
                 function(success) {
-                    $scope.csvimportcounter++;
+                    $state.go('users.user.list');
                 }
             );
-        }
-        $scope.csvimported = true;
-    };
+        };
+    }
+])
 
-    $scope.clear = function () {
-        $scope.csv.result = null;
-    };
-})
+.controller('UserProfileCtrl', [
+    '$scope',
+    '$state',
+    'User',
+    'user',
+    function($scope, $state, User, user) {
+        $scope.user = user;  // autoupdate is not activated
+        $scope.save = function (user) {
+            User.save(user).then(
+                function(success) {
+                    $state.go('users.user.list');
+                },
+                function(error) {
+                    $scope.formError = error;
+                }
+            );
+        };
+    }
+])
 
-.controller('GroupListCtrl', function($scope, Group) {
-    Group.bindAll({}, $scope, 'groups');
+.controller('UserPasswordCtrl', [
+    '$scope',
+    '$state',
+    '$http',
+    'user',
+    function($scope, $state, $http, user) {
+        $scope.user = user;  // autoupdate is not activated
+        $scope.save = function (user) {
+            if ($scope.newPassword != $scope.newPassword2) {
+                $scope.newPassword = $scope.newPassword2 = '';
+                $scope.formError = 'Password confirmation does not match.'
+            } else {
+                $http.post(
+                    '/users/setpassword/',
+                    {'old_password': $scope.oldPassword, 'new_password': $scope.newPassword}
+                ).then(
+                    function(data) {
+                        $state.go('users.user.list');
+                    },
+                    function(data) {
+                        $scope.oldPassword = $scope.newPassword = $scope.newPassword2 = ''
+                        $scope.formError = data;
+                    }
+                );
```

Comment:
I added a comment. This case is solved in line 450 and 451. ValidationError returns HTTP 400.